csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -1076,7 +1086,8 @@ private Reference CreateReference(Identity identity, string facet, string adapte
                                      facet: facet,
                                      identity: identity,
                                      invocationMode: _invocationMode,
-                                     protocol: Protocol);
+                                     protocol: adapterId.Length > 0 ?
+                                               Protocol : _publishedEndpoints.FirstOrDefault()?.Protocol ?? Protocol);
```

Comment:
> When adapterId is set, the OA cannot have published endpoints?
It can but they are not used in the Reference
```
endpoints: adapterId.Length == 0 ? _publishedEndpoints : Array.Empty<Endpoint>(),
```

> Is this adapterId.Length check correct?
I think it is not, even if the reference has no endpoints, in this case, we want to use the Protocol of the published endpoints.